app.rb

```diff
@@ -55,6 +55,17 @@
     elsif !BSON::ObjectId.legal?(@token)
       puts 'haha'
       halt(401, "Invalid Token")
+    elsif request.path_info.include? 'admin'
+      puts "Checking admin credentials"
+      session = collection.find( {:_id => BSON::ObjectId(@token) }).first
+      if session.nil?
+        halt(401, "Invalid Token in admin check")
```

Comment:
Lines 60-62 are basically the same as lines 70-73. See if you can find a way to consolidate that functionality and eliminate duplicate code. Check out the Helpers section of the intro docs: http://sinatrarb.com/intro.html.